trl/rewards/accuracy_rewards.py

```diff
@@ -70,10 +74,119 @@ def accuracy_reward(completions: list[list[dict[str, str]]], solution: list[str]
                 extraction_mode="first_match",
             )
             # Compute binary rewards if verifiable, `None` otherwise to skip this example
-            reward = float(verify(gold_parsed, answer_parsed))
+            try:
+                reward = float(verify(gold_parsed, answer_parsed))
+            except Exception as e:
+                logger.warning(f"Verification failed with error: {e}. Assigning reward=None for this example.")
+                reward = None
+        else:
+            # If the gold solution cannot be parsed, we assign `None` to skip this example
+            reward = None
+        rewards.append(reward)
+
+    return rewards
+
+
+def reasoning_accuracy_reward(
```

Comment:
We could technically merge this logic with `accuracy_reward()`, but for now I kept it separate to avoid bundling too many things together. Happy to refactor if you prefer it all in one function.

Reply:
lgtm!